### ISSUE TITLE
大图跑graphsage时failed

### ISSUE THREAD

**Issue by Post-Dawn** [Angel-ML/angel#1048]
运行脚本如下：
source ./spark-on-angel-env.sh
$SPARK_HOME/bin/spark-submit \
      --master yarn-cluster\
      --conf spark.ps.instances=2 \
      --conf spark.ps.cores=10 \
      --conf spark.ps.jars=$SONA_ANGEL_JARS \
      --conf spark.ps.memory=8g \
      --conf spark.ps.log.level=INFO \
      --conf spark.driver.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib \
      --conf spark.executor.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib \
      --conf spark.executor.extraLibraryPath=$TORCH_HOME/lib \
      --conf spark.driver.extraLibraryPath=$TORCH_HOME/lib \
      --conf spark.executorEnv.OMP_NUM_THREADS=2 \
      --conf spark.executorEnv.MKL_NUM_THREADS=2 \
      --queue default \
      --name "graphsage-angel" \
      --jars $SONA_SPARK_JARS  \
      --archives hdfs:///torch.zip \
      --files ./graphsage_cora.pt \
      --driver-memory 5g \
      --num-executors 3 \
      --executor-cores 5 \
      --executor-memory 5g \
      --class com.tencent.angel.pytorch.examples.supervised.GCNExample \
      ../pytorch-on-angel-*.jar \
      edgePath:hdfs:///now/input/edge.csv featurePath:hdfs:///now/input/feature.csv labelPath:hdfs:///now/input/label.csv \
      torchModelPath:graphsage_cora.pt featureDim:1 stepSize:0.01 \
      optimizer:adam numEpoch:20 testRatio:0.5 \
      numPartitions:50 format:dense samples:10 batchSize:128 \
      predictOutputPath:hdfs:///now/output/ embeddingPath:hdfs:///now/embedding/ outputModelPath:hdfs:///now/model/ \
      actionType:train numBatchInit:5
在angel的log中主要报错：
2020-11-23 15:16:34,826 FATAL [app-state-monitor] com.tencent.angel.master.app.InternalErrorEvent: app in state NEW over 3000000 milliseconds!
边大概在30000000左右

**Comment by ouyangwen-it** [comment]
> 运行脚本如下：
> source ./spark-on-angel-env.sh
> $SPARK_HOME/bin/spark-submit
> --master yarn-cluster
> --conf spark.ps.instances=2
> --conf spark.ps.cores=10
> --conf spark.ps.jars=$SONA_ANGEL_JARS
> --conf spark.ps.memory=8g
> --conf spark.ps.log.level=INFO
> --conf spark.driver.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> --conf spark.executor.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> --conf spark.executor.extraLibraryPath=$TORCH_HOME/lib
> --conf spark.driver.extraLibraryPath=$TORCH_HOME/lib
> --conf spark.executorEnv.OMP_NUM_THREADS=2
> --conf spark.executorEnv.MKL_NUM_THREADS=2
> --queue default
> --name "graphsage-angel"
> --jars $SONA_SPARK_JARS
> --archives hdfs:///torch.zip
> --files ./graphsage_cora.pt
> --driver-memory 5g
> --num-executors 3
> --executor-cores 5
> --executor-memory 5g
> --class com.tencent.angel.pytorch.examples.supervised.GCNExample
> ../pytorch-on-angel-*.jar
> edgePath:hdfs:///now/input/edge.csv featurePath:hdfs:///now/input/feature.csv labelPath:hdfs:///now/input/label.csv
> torchModelPath:graphsage_cora.pt featureDim:1 stepSize:0.01
> optimizer:adam numEpoch:20 testRatio:0.5
> numPartitions:50 format:dense samples:10 batchSize:128
> predictOutputPath:hdfs:///now/output/ embeddingPath:hdfs:///now/embedding/ outputModelPath:hdfs:///now/model/
> actionType:train numBatchInit:5
> 在angel的log中主要报错：
> 2020-11-23 15:16:34,826 FATAL [app-state-monitor] com.tencent.angel.master.app.InternalErrorEvent: app in state NEW over 3000000 milliseconds!
> 边大概在30000000左右

![image](https://user-images.githubusercontent.com/13943417/100055130-54d54a80-2e5e-11eb-9ed3-685ace1df1ab.png)
看日志app处在new状态都没运行，应该是申请不到资源，请确定你集群或者队列有足够的资源满足你的申请

**Comment by Post-Dawn** [comment]
> > 运行脚本如下：
> > source ./spark-on-angel-env.sh
> > $SPARK_HOME/bin/spark-submit
> > --master yarn-cluster
> > --conf spark.ps.instances=2
> > --conf spark.ps.cores=10
> > --conf spark.ps.jars=$SONA_ANGEL_JARS
> > --conf spark.ps.memory=8g
> > --conf spark.ps.log.level=INFO
> > --conf spark.driver.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> > --conf spark.executor.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> > --conf spark.executor.extraLibraryPath=$TORCH_HOME/lib
> > --conf spark.driver.extraLibraryPath=$TORCH_HOME/lib
> > --conf spark.executorEnv.OMP_NUM_THREADS=2
> > --conf spark.executorEnv.MKL_NUM_THREADS=2
> > --queue default
> > --name "graphsage-angel"
> > --jars $SONA_SPARK_JARS
> > --archives hdfs:///torch.zip
> > --files ./graphsage_cora.pt
> > --driver-memory 5g
> > --num-executors 3
> > --executor-cores 5
> > --executor-memory 5g
> > --class com.tencent.angel.pytorch.examples.supervised.GCNExample
> > ../pytorch-on-angel-*.jar
> > edgePath:hdfs:///now/input/edge.csv featurePath:hdfs:///now/input/feature.csv labelPath:hdfs:///now/input/label.csv
> > torchModelPath:graphsage_cora.pt featureDim:1 stepSize:0.01
> > optimizer:adam numEpoch:20 testRatio:0.5
> > numPartitions:50 format:dense samples:10 batchSize:128
> > predictOutputPath:hdfs:///now/output/ embeddingPath:hdfs:///now/embedding/ outputModelPath:hdfs:///now/model/
> > actionType:train numBatchInit:5
> > 在angel的log中主要报错：
> > 2020-11-23 15:16:34,826 FATAL [app-state-monitor] com.tencent.angel.master.app.InternalErrorEvent: app in state NEW over 3000000 milliseconds!
> > 边大概在30000000左右
> 
> ![image](https://user-images.githubusercontent.com/13943417/100055130-54d54a80-2e5e-11eb-9ed3-685ace1df1ab.png)
> 看日志app处在new状态都没运行，应该是申请不到资源，请确定你集群或者队列有足够的资源满足你的申请

我报错的时候集群有40GB的内存，我算了下总共消耗了36GB。
虽然我把集群改成90GB就通过了，不过应该之前的时候资源应该是够的？

**Comment by ouyangwen-it** [comment]
> > > 运行脚本如下：
> > > source ./spark-on-angel-env.sh
> > > $SPARK_HOME/bin/spark-submit
> > > --master yarn-cluster
> > > --conf spark.ps.instances=2
> > > --conf spark.ps.cores=10
> > > --conf spark.ps.jars=$SONA_ANGEL_JARS
> > > --conf spark.ps.memory=8g
> > > --conf spark.ps.log.level=INFO
> > > --conf spark.driver.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> > > --conf spark.executor.extraJavaOptions=-Djava.library.path=../:$TORCH_HOME/lib
> > > --conf spark.executor.extraLibraryPath=$TORCH_HOME/lib
> > > --conf spark.driver.extraLibraryPath=$TORCH_HOME/lib
> > > --conf spark.executorEnv.OMP_NUM_THREADS=2
> > > --conf spark.executorEnv.MKL_NUM_THREADS=2
> > > --queue default
> > > --name "graphsage-angel"
> > > --jars $SONA_SPARK_JARS
> > > --archives hdfs:///torch.zip
> > > --files ./graphsage_cora.pt
> > > --driver-memory 5g
> > > --num-executors 3
> > > --executor-cores 5
> > > --executor-memory 5g
> > > --class com.tencent.angel.pytorch.examples.supervised.GCNExample
> > > ../pytorch-on-angel-*.jar
> > > edgePath:hdfs:///now/input/edge.csv featurePath:hdfs:///now/input/feature.csv labelPath:hdfs:///now/input/label.csv
> > > torchModelPath:graphsage_cora.pt featureDim:1 stepSize:0.01
> > > optimizer:adam numEpoch:20 testRatio:0.5
> > > numPartitions:50 format:dense samples:10 batchSize:128
> > > predictOutputPath:hdfs:///now/output/ embeddingPath:hdfs:///now/embedding/ outputModelPath:hdfs:///now/model/
> > > actionType:train numBatchInit:5
> > > 在angel的log中主要报错：
> > > 2020-11-23 15:16:34,826 FATAL [app-state-monitor] com.tencent.angel.master.app.InternalErrorEvent: app in state NEW over 3000000 milliseconds!
> > > 边大概在30000000左右
> > 
> > 
> > ![image](https://user-images.githubusercontent.com/13943417/100055130-54d54a80-2e5e-11eb-9ed3-685ace1df1ab.png)
> > 看日志app处在new状态都没运行，应该是申请不到资源，请确定你集群或者队列有足够的资源满足你的申请
> 
> 我报错的时候集群有40GB的内存，我算了下总共消耗了36GB。
> 虽然我把集群改成90GB就通过了，不过应该之前的时候资源应该是够的？（虽然肯定是资源的问题

还要看下core资源是不是够的，另外需要加上angel ps master需要的资源(默认是4g，1个core)
